core/sdk-core/src/test/java/software/amazon/awssdk/core/client/builder/DefaultClientBuilderTest.java

```diff
@@ -267,6 +276,7 @@ public void close() {
         assertThat(config.option(METRIC_PUBLISHERS)).contains(metricPublisher);
         assertThat(config.option(EXECUTION_ATTRIBUTES).getAttribute(execAttribute)).isEqualTo("value");
         assertThat(config.option(ENDPOINT_OVERRIDDEN)).isEqualTo(Boolean.TRUE);
+        assertThat(config.option(SCHEDULED_EXECUTOR_SERVICE)).isEqualTo(scheduledExecutorService);
```

Comment:
Now that we are wrapping with `NonManagedScheduledExecutorService`, we need to retrieve the custom `ScheduledExecutorService`

```
import static software.amazon.awssdk.core.client.builder.SdkDefaultClientBuilder.NonManagedScheduledExecutorService;

        NonManagedScheduledExecutorService customScheduledExecutorService =
            (NonManagedScheduledExecutorService) config.option(SCHEDULED_EXECUTOR_SERVICE);
        assertThat(customScheduledExecutorService.scheduledExecutorService()).isEqualTo(scheduledExecutorService);
```